From d1d686cb5cebca571cb9eafc5a3cabff6971ddb6 Mon Sep 17 00:00:00 2001
From: Ariel D'Alessandro <ariel.dalessandro@collabora.com>
Date: Sat, 26 Apr 2025 17:19:24 -0300
Subject: [PATCH] rust: Use adler instead of adler2

Upstream Rust replaced adler with adler2, for older versions of Rust we still
need to tell GN that we have the older lib when it tries to copy the Rust
sysroot into the bulid directory.

It looks that in rust-1.85.0, `adler` has not yet been replaced with `adler2`.
Therefore, the condition should likely be adjusted.

This patch will be dropped once Rust >= 1.86.0 is available, which
replaced adler with adler2.

Note that currently layer provides Rust version `1.85.1`.

[0] https://git.yoctoproject.org/meta-lts-mixins/log/?h=scarthgap/rust

Upstream-Status: Inappropriate [specific to older versions of rust]
Signed-off-by: Ariel D'Alessandro <ariel.dalessandro@collabora.com>
---
 build/rust/std/BUILD.gn | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build/rust/std/BUILD.gn b/build/rust/std/BUILD.gn
index 7f64198b9f..d29e3466b1 100644
--- a/build/rust/std/BUILD.gn
+++ b/build/rust/std/BUILD.gn
@@ -89,7 +89,7 @@ if (toolchain_has_rust) {
     # These are no longer present in the Windows toolchain.
     stdlib_files += [
       "addr2line",
-      "adler2",
+      "adler",
       "gimli",
       "libc",
       "memchr",
